source/main_cr.f90

```diff
@@ -1240,275 +1241,228 @@ end subroutine envarsv
-          idam=idam-1
-        else
-          icode=icode+1
-        endif
-        if(abs(xv2(ia)).le.pieni.and.abs(yv2(ia)).le.pieni) then
-          idam=idam-1
-        else
-          icode=icode+2
-        endif
-        if(idp.eq.0.or.abs(ition).eq.0) then
-          idam=idam-1
-        else
-          icode=icode+4
-        endif
-        if(idam.le.0) idam=1
-        if(icode.le.0) icode=1
-        ia2=(ia+1)/2
-        if(ntwin.ne.2) then
-          if(mod(ia+1,2).eq.0) then
-            xau(1,1)= xv1(ia)
-            xau(1,2)= yv1(ia)
-            xau(1,3)= xv2(ia)
-            xau(1,4)= yv2(ia)
-            xau(1,5)=sigmv(ia)
-            xau(1,6)= dpsv(ia)
-            xau(2,1)= xv1(ia+1)
-            xau(2,2)= yv1(ia+1)
-            xau(2,3)= xv2(ia+1)
-            xau(2,4)= yv2(ia+1)
-            xau(2,5)=sigmv(ia+1)
-            xau(2,6)= dpsv(ia+1)
-            cloau(1)= clo6v(1,ia)
-            cloau(2)=clop6v(1,ia)
-            cloau(3)= clo6v(2,ia)
-            cloau(4)=clop6v(2,ia)
-            cloau(5)= clo6v(3,ia)
-            cloau(6)=clop6v(3,ia)
-            di0au(1)= di0xs(ia)
-            di0au(2)=dip0xs(ia)
-            di0au(3)= di0zs(ia)
-            di0au(4)=dip0zs(ia)
+        if(abs(bet0s1) <= pieni) x2(6) = dpsv(ia)
+        if(iver == 1) then
+          x2(3) = zero
+          x2(4) = zero
+        end if
+        xv1(i3)   = x2(1)+exz(i2,1)
+        yv1(i3)   = x2(2)+exz(i2,2)
+        xv2(i3)   = x2(3)+exz(i2,3)
+        yv2(i3)   = x2(4)+exz(i2,4)
+        sigmv(i3) = x2(5)+exz(i2,5)
+        dpsv(i3)  = x2(6)
+        dpsic     = dpsv(i3)+clop6v(3,ia)
+        if(idp == 1 .and. abs(ition) == 1 .and. iclo6 == 0) then
+          xv1(i3) = xv1(i3) + di0xs(ia)*dpsic
+          xv2(i3) = xv2(i3) + di0zs(ia)*dpsic
+          yv1(i3) = yv1(i3) + dip0xs(ia)*dpsic
+          yv2(i3) = yv2(i3) + dip0zs(ia)*dpsic
+        end if
+        chi = chi+dchi
+      end do
 
-            do ib2=1,6
-              do ib3=1,6
-                tau(ib2,ib3)=tasau(ia,ib2,ib3)
-              end do
-            end do
+      epsa(1)    = (ampv(ia)**2/bet0v(ia,1))
+      epsa(2)    = (amp(2)**2/bet0v(ia,2))
+      nucm(ia)   = nucm0
+      nucm(ia+1) = nucm0
 
-            call distance(xau,cloau,di0au,tau,dam1)
-            dam(ia)=dam1
-            dam(ia+1)=dam1
-          endif !endif(mod(ia+1,2).eq.0)
+      if(st_quiet == 0) then
+        write(lout,10260) ia,nms(ia)*izu0,dpsv(ia)
+        write(lout,10060) xv1(ia),yv1(ia),xv2(ia),yv2(ia),sigmv(ia),dpsv(ia), &
+          xv1(ia+1),yv1(ia+1),xv2(ia+1),yv2(ia+1),sigmv(ia+1),dpsv(ia+1)
+        write(lout,10020) ampv(ia),amp(2),epsa
+      end if
+    end do
+    call part_applyClosedOrbit
+
+  else if(idfor == 2) then
+    ! Read from fort.13
+    call readFort13
+    call part_updatePartEnergy(3)
+    ! Note that this effectively overrides the particle energy set in fort.13
+    ! as energy is recalculated from delta.
+  endif
+
```

Comment:
I would add here an explicit `call part_updatePartEnergy(3)` instead of having it in `part_applyClosedOrbit` (or ad-hoc if `idfor==2`). Before, we could even add a unique `call part_applyClosedOrbit` under condition `idfor!=2`.

Reply:
The calls are as they are to ensure the canoncals don't change for the tests using the INIT block (most of them). It was tricky to make this work and I am reluctant to start shuffling things around again.

Reply:
ok - let's wait for such a cleanup once we are able to re-validate all tests following code modifications which entangle changes in the last digits of results... do you agree, @rdemaria ?